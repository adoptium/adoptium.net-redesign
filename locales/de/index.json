{
  "Prebuilt OpenJDK Binaries for Free!": "Kostenlose OpenJDK™ Binaries zum Download!",
  "Intro": "Java™ ist die weltweit führende Programmiersprache und Softwareplattform. Die Adoptium Working Group fördert und unterstützt qualitativ hochwertige, TCK-zertifizierte Laufzeitumgebungen und zugehörige Technologien für den Einsatz im gesamten Java-Ökosystem. Eclipse Temurin ist der Name der OpenJDK-Distribution von Adoptium."
}
