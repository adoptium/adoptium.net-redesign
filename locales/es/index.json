{
  "Prebuilt OpenJDK Binaries for Free!": "OpenJDK Preconstruido ¡Binarios Gratis!",
  "Intro": "Java™ es el lenguaje de programación y la plataforma líderes en el mundo. El Grupo de Trabajo de Adoptium promueve y admite tiempos de ejecución de alta calidad, certificados por TCK y tecnología asociada para su uso en todo el ecosistema Java. Eclipse Temurin es el nombre de la distribución OpenJDK de Adoptium."
}
