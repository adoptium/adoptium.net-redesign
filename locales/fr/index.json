{
  "Prebuilt OpenJDK Binaries for Free!": "OpenJDK précompilé Des binaires gratuits !",
  "Intro": "Java™ est le premier langage de programmation et la première plate-forme de développement au monde. Le groupe de travail Adoptium supporte et produit des environnements d'exécution de haute qualité, certifiés TCK ainsi que les technologies associées, à utiliser dans l'ensemble de l'écosystème Java. Eclipse Temurin est le nom de la distribution OpenJDK d'Adoptium."
}
