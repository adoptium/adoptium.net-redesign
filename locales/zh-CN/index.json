{
  "Prebuilt OpenJDK Binaries for Free!": "预编译 OpenJDK 免费下载！",
  "Intro": "Java™ 是世界领先的编程语言和平台。Adoptium 工作推进和支持高质量、TCK 认证的运行时和其相关技术，使其在 Java 生态系统中应用。Eclipse Temurin 是 Adoptium OpenJDK 发行版的名称。"
}
