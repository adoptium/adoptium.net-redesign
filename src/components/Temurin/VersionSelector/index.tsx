import React, { useCallback } from "react"

import { setURLParam } from "../../../util/setURLParam"

const VersionSelector = ({
  activeVersionTab,
  setActiveVersionTab,
  versions,
  updateVersion,
  defaultVersion,
}) => {
  const setActiveTabVersion = useCallback(newActiveVersionTab => {
    if (newActiveVersionTab === 1) {
      setURLParam("version", defaultVersion)
      updateVersion(defaultVersion)
    } else {
      setURLParam("version", newActiveVersionTab)
      updateVersion(newActiveVersionTab)
    }
    setActiveVersionTab(newActiveVersionTab)
  }, [])

  return (
    <div className="w-full max-w-[1264px] mx-auto ">
      <div className="overflow-auto relative min-w-full w-full ">
        <span className="h-[1px] w-full bg-[#3E3355] inline-block absolute bottom-0 z-[-1]"></span>
        <div className="flex space-x-10 whitespace-nowrap   lg:justify-center py-3">
          <button onClick={() => setActiveTabVersion(1)}>
            <span
              className={` py-3  w-full text-base font-normal leading-6 
                outline-none cursor-pointer transition-all duration-200 ease-in-out ${activeVersionTab === 1 ? "border-primary  border-b-[2px] text-white" : "text-[#8a809e] border-transparent  border-b"}`}
            >
              All Versions
            </span>
          </button>
          {versions.map((version, index) => (
            <button
              key={index}
              onClick={() => setActiveTabVersion(version.node.version)}
            >
              <span
                className={` py-3  w-full text-base font-normal leading-6
                    outline-none cursor-pointer transition-all duration-200 ease-in-out ${activeVersionTab === version.node.version ? "border-primary  border-b-[2px] text-white" : "text-[#8a809e] border-transparent  border-b"}`}
              >
                JDK {version.node.label}
              </span>
            </button>
          ))}
        </div>
      </div>
    </div>
  )
}

export default VersionSelector
