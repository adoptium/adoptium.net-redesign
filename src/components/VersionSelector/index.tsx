import React, { useState, useCallback, useEffect } from "react"
import { useStaticQuery, graphql } from "gatsby"
import { Trans, useI18next } from "gatsby-plugin-react-i18next"
import { useLocation } from "@gatsbyjs/reach-router"
import queryString from "query-string"
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV3'
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider"
import { DesktopDatePicker } from "@mui/x-date-pickers/DesktopDatePicker"
import * as Locales from 'date-fns/locale'

import { setURLParam } from "../../util/setURLParam"

const VersionSelector = ({ updater, releaseType, Table }) => {
  const data = useStaticQuery(graphql`
    query VersionsQuery {
      allVersions(sort: { version: DESC }) {
        edges {
          node {
            version
            label
            lts
          }
        }
      }
      mostRecentLts {
        version
      }
      mostRecentFeatureVersion {
        version
      }
    }
  `)

  const defaultVersion = data.mostRecentLts.version
  const mostRecentFeatureVersion = data.mostRecentFeatureVersion.version
  const versions = data.allVersions.edges

  const { language } = useI18next()

  let locale

  // import the correct date locale for the language
  if (language !== "en") {
    locale = Locales[language] ?? Locales[language.substring(0, 2)] ?? Locales.enUS
  }

  let selectedVersion = defaultVersion
  const versionParam = queryString.parse(useLocation().search).version
  if (versionParam) {
    selectedVersion = Number(versionParam)
  }
  const variantParam = queryString.parse(useLocation().search).variant
  if (variantParam) {
    // convert openjdk11 to 11
    const parsedVersion = variantParam.toString().replace(/\D/g, "")
    setURLParam("version", parsedVersion)
    selectedVersion = parseInt(parsedVersion)
  }

  const [version, udateVersion] = useState(selectedVersion.toString())
  const [numBuilds, udateNumBuilds] = useState(5)
  const [page, updatePage] = useState(0)
  const [buildDate, updateBuildDate] = useState<Date>(new Date())
  const [releases, setReleases] = useState(null)

  useEffect(() => {
    (async () => {
      setReleases(
        await updater(version, releaseType, numBuilds, buildDate, page),
      )
    })()
  }, [version, numBuilds, buildDate, page])

  const setVersion = useCallback(version => {
    setURLParam("version", version)
    udateVersion(version)
  }, [])

  const setNumBuilds = useCallback(number => {
    udateNumBuilds(number)
  }, [])

  return (
    <>
      <p className="text-center">
        <Trans>
          Use the drop-down boxes below to filter the list of releases.
        </Trans>
      </p>
      <div className="input-group p-3 d-flex justify-content-center">
        <label className="px-2 fw-bold" htmlFor="version">
          <Trans>Version</Trans>
        </label>
        <select
          data-testid="version-filter"
          aria-label="version-filter"
          id="version-filter"
          onChange={e => setVersion(e.target.value)}
          value={version}
          className="form-select form-select-sm"
          style={{ maxWidth: "10em" }}
        >
          {/* if releaseType is ea add another option */}
          {releaseType === "ea" && (
            <option
              key={mostRecentFeatureVersion}
              value={mostRecentFeatureVersion}
            >{`${mostRecentFeatureVersion} - EA`}</option>
          )}
          {/* loop through versions array from graphql */}
          {versions.map(
            (version, i): number | JSX.Element =>
              version && (
                <option key={version.node.id} value={version.node.version}>
                  {version.node.label}
                </option>
              ),
          )}
        </select>
      </div>
      {releaseType === "ea" && (
        <div className="input-group pb-5 d-flex justify-content-center">
          <span className="p-2">View</span>
          <select
            data-testid="build-num-filter"
            aria-label="Filter by number of builds"
            id="build-num-filter"
            onChange={e => setNumBuilds(e.target.value)}
            defaultValue={numBuilds}
            className="form-select form-select-sm"
            style={{ maxWidth: "5em" }}
          >
            <option key={1} value={1}>
              1
            </option>
            <option key={5} value={5}>
              5
            </option>
            <option key={10} value={10}>
              10
            </option>
            <option key={20} value={20}>
              20
            </option>
            <option key={50} value={50}>
              50
            </option>
          </select>
          <span className="p-2">nightly builds prior to:</span>
          <LocalizationProvider
            dateAdapter={AdapterDateFns}
            adapterLocale={locale}
          >
            <DesktopDatePicker
              label="Build Date"
              value={buildDate}
              maxDate={new Date()}
              onChange={newValue => {
                newValue && updateBuildDate(newValue)
              }}
              sx={{
                maxWidth: "10em",
              }}
            />
          </LocalizationProvider>
        </div>
      )}
      <Table results={releases} updatePage={updatePage} />
    </>
  )
}

export default VersionSelector
