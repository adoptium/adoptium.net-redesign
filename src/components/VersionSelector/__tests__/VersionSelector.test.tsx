import React from "react"
import {
  act,
  render,
  screen,
  fireEvent,
  prettyDOM,
} from "@testing-library/react"
import "@testing-library/jest-dom"
import queryString from "query-string"
import { afterEach, vi, describe, it, expect } from "vitest"
import { useI18next } from "gatsby-plugin-react-i18next"
import VersionSelector from "../index"
import locales from "../../../../locales/i18n"

const defaultVersion = 1

describe("VersionSelector", () => {
  const updater = vi.fn()
  const releaseType = "ga"
  const Table = () => <div>Table</div>

  vi.mock("query-string")

  vi.mock("gatsby-plugin-react-i18next")

  // @ts-ignore
  useI18next.mockReturnValue({
    language: "en",
    languages: locales,
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  it("renders the component with default values", async () => {
    queryString.parse = vi.fn().mockReturnValue({})
    await act(async () => {
      render(
        <VersionSelector
          updater={updater}
          releaseType={releaseType}
          Table={Table}
        />,
      )
    })
    expect(screen.getByTestId("version-filter")).toHaveValue(
      defaultVersion.toString(),
    )
    expect(screen.queryByLabelText("View")).not.toBeInTheDocument()
    expect(
      screen.queryByLabelText("nightly builds prior to:"),
    ).not.toBeInTheDocument()
    expect(screen.getByText("Table")).toBeInTheDocument()
  })

  it('renders the component with releaseType "ea"', async () => {
    queryString.parse = vi.fn().mockReturnValue({})
    await act(async () => {
      render(
        <VersionSelector updater={updater} releaseType="ea" Table={Table} />,
      )
    })
    expect(screen.getByTestId("version-filter")).toHaveValue(
      defaultVersion.toString(),
    )
    expect(screen.getByTestId("build-num-filter")).toHaveValue("5")
    expect(screen.getByText("Table")).toBeInTheDocument()
  })

  it("updates the version when the select input changes", async () => {
    queryString.parse = vi.fn().mockReturnValue({})
    render(
      <VersionSelector
        updater={updater}
        releaseType={releaseType}
        Table={Table}
      />,
    )
    await act(async () => {
      fireEvent.change(screen.getByTestId("version-filter"), {
        target: { value: "2" },
      })
    })
    expect(updater).toHaveBeenCalledWith(
      "2",
      releaseType,
      5,
      expect.any(Date),
      0,
    )
  })

  it("updates the number of builds and build date when the inputs change", async () => {
    queryString.parse = vi.fn().mockReturnValue({})
    let container
    await act(async () => {
      const result = render(
        <VersionSelector updater={updater} releaseType="ea" Table={Table} />,
      )
      container = result.container
    })

    fireEvent.change(screen.getByTestId("build-num-filter"), {
      target: { value: 10 },
    })

    await act(async () => {
      const datepicker = screen.getByLabelText("Build Date")
      fireEvent.change(datepicker, { target: { value: "2022-01-01" } })
      expect(datepicker.getAttribute("value")).toBe("2022-01-01")
    })

    expect(updater).lastCalledWith("1", "ea", "10", expect.any(Date), 0)

    // Add the snapshot test for the final rendered output
    expect(prettyDOM(container)).toMatchSnapshot()
  })

  it("renders the component with version query param", async () => {
    queryString.parse = vi.fn().mockReturnValue({ version: 2 })

    await act(async () => {
      render(
        <VersionSelector
          updater={updater}
          releaseType={releaseType}
          Table={Table}
        />,
      )
    })
    expect(screen.getByTestId("version-filter")).toHaveValue("2")
    expect(screen.queryByLabelText("View")).not.toBeInTheDocument()
    expect(
      screen.queryByLabelText("nightly builds prior to:"),
    ).not.toBeInTheDocument()
    expect(screen.getByText("Table")).toBeInTheDocument()
  })

  it("renders the component with variant query param", async () => {
    queryString.parse = vi.fn().mockReturnValue({ variant: "openjdk2" })

    await act(async () => {
      render(
        <VersionSelector
          updater={updater}
          releaseType={releaseType}
          Table={Table}
        />,
      )
    })
    expect(screen.getByTestId("version-filter")).toHaveValue("2")
    expect(screen.queryByLabelText("View")).not.toBeInTheDocument()
    expect(
      screen.queryByLabelText("nightly builds prior to:"),
    ).not.toBeInTheDocument()
    expect(screen.getByText("Table")).toBeInTheDocument()
  })

  // loop through locales
  Object.keys(locales).forEach(locale => {
    it(`renders when the locale is set to ${locale}`, async () => {
      queryString.parse = vi.fn().mockReturnValue({})

      // @ts-ignore
      useI18next.mockReturnValue({
        language: locale,
        languages: locales,
      })

      await act(async () => {
        render(
          <VersionSelector updater={updater} releaseType="ea" Table={Table} />,
        )
      })
      const datepicker = screen.getByLabelText("Build Date")
      expect(datepicker).toBeInTheDocument()
    })
  })
})
