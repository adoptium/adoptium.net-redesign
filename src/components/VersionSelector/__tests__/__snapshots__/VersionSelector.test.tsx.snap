// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VersionSelector > updates the number of builds and build date when the inputs change 1`] = `
"[36m<div>[39m
  [36m<p[39m
    [33mclass[39m=[32m"text-center"[39m
  [36m/>[39m
  [36m<div[39m
    [33mclass[39m=[32m"input-group p-3 d-flex justify-content-center"[39m
  [36m>[39m
    [36m<label[39m
      [33mclass[39m=[32m"px-2 fw-bold"[39m
      [33mfor[39m=[32m"version"[39m
    [36m/>[39m
    [36m<select[39m
      [33maria-label[39m=[32m"version-filter"[39m
      [33mclass[39m=[32m"form-select form-select-sm"[39m
      [33mdata-testid[39m=[32m"version-filter"[39m
      [33mid[39m=[32m"version-filter"[39m
      [33mstyle[39m=[32m"max-width: 10em;"[39m
    [36m>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"24"[39m
      [36m>[39m
        [0m24 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"23"[39m
      [36m>[39m
        [0m23 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"22"[39m
      [36m>[39m
        [0m22 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"21"[39m
      [36m>[39m
        [0m21 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"20"[39m
      [36m>[39m
        [0m20 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"19"[39m
      [36m>[39m
        [0m19 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"18"[39m
      [36m>[39m
        [0m18 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"17"[39m
      [36m>[39m
        [0m17 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"16"[39m
      [36m>[39m
        [0m16 - EA[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"2"[39m
      [36m>[39m
        [0m2[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"1"[39m
      [36m>[39m
        [0m1 - LTS[0m
      [36m</option>[39m
    [36m</select>[39m
  [36m</div>[39m
  [36m<div[39m
    [33mclass[39m=[32m"input-group pb-5 d-flex justify-content-center"[39m
  [36m>[39m
    [36m<span[39m
      [33mclass[39m=[32m"p-2"[39m
    [36m>[39m
      [0mView[0m
    [36m</span>[39m
    [36m<select[39m
      [33maria-label[39m=[32m"Filter by number of builds"[39m
      [33mclass[39m=[32m"form-select form-select-sm"[39m
      [33mdata-testid[39m=[32m"build-num-filter"[39m
      [33mid[39m=[32m"build-num-filter"[39m
      [33mstyle[39m=[32m"max-width: 5em;"[39m
    [36m>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"1"[39m
      [36m>[39m
        [0m1[0m
      [36m</option>[39m
      [36m<option[39m
        [33mselected[39m=[32m""[39m
        [33mvalue[39m=[32m"5"[39m
      [36m>[39m
        [0m5[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"10"[39m
      [36m>[39m
        [0m10[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"20"[39m
      [36m>[39m
        [0m20[0m
      [36m</option>[39m
      [36m<option[39m
        [33mvalue[39m=[32m"50"[39m
      [36m>[39m
        [0m50[0m
      [36m</option>[39m
    [36m</select>[39m
    [36m<span[39m
      [33mclass[39m=[32m"p-2"[39m
    [36m>[39m
      [0mnightly builds prior to:[0m
    [36m</span>[39m
    [36m<div[39m
      [33mclass[39m=[32m"MuiFormControl-root MuiTextField-root css-b8bldw-MuiFormControl-root-MuiTextField-root"[39m
    [36m>[39m
      [36m<label[39m
        [33mclass[39m=[32m"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"[39m
        [33mdata-shrink[39m=[32m"true"[39m
        [33mfor[39m=[32m":r3:"[39m
        [33mid[39m=[32m":r3:-label"[39m
      [36m>[39m
        [0mBuild Date[0m
      [36m</label>[39m
      [36m<div[39m
        [33mclass[39m=[32m"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"[39m
      [36m>[39m
        [36m<input[39m
          [33maria-invalid[39m=[32m"false"[39m
          [33mautocomplete[39m=[32m"off"[39m
          [33mclass[39m=[32m"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"[39m
          [33mid[39m=[32m":r3:"[39m
          [33minputmode[39m=[32m"text"[39m
          [33mplaceholder[39m=[32m"MM/DD/YYYY"[39m
          [33mtype[39m=[32m"text"[39m
          [33mvalue[39m=[32m"01/01/2022"[39m
        [36m/>[39m
        [36m<div[39m
          [33mclass[39m=[32m"MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"[39m
        [36m>[39m
          [36m<button[39m
            [33maria-label[39m=[32m"Choose date, selected date is Jan 1, 2022"[39m
            [33mclass[39m=[32m"MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"[39m
            [33mtabindex[39m=[32m"0"[39m
            [33mtype[39m=[32m"button"[39m
          [36m>[39m
            [36m<svg[39m
              [33maria-hidden[39m=[32m"true"[39m
              [33mclass[39m=[32m"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"[39m
              [33mdata-testid[39m=[32m"CalendarIcon"[39m
              [33mfocusable[39m=[32m"false"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"[39m
              [36m/>[39m
            [36m</svg>[39m
          [36m</button>[39m
        [36m</div>[39m
        [36m<fieldset[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"[39m
        [36m>[39m
          [36m<legend[39m
            [33mclass[39m=[32m"css-w1u3ce"[39m
          [36m>[39m
            [36m<span>[39m
              [0mBuild Date[0m
            [36m</span>[39m
          [36m</legend>[39m
        [36m</fieldset>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
  [36m<div>[39m
    [0mTable[0m
  [36m</div>[39m
[36m</div>[39m"
`;
